<!DOCTYPE html>
<html lang="en" class="no-js">
  <!-- Copyright 2019-2021 Vanessa Sochat-->
  <head>
<meta charset="utf-8">
<!-- Copyright 2019-2021 Vanessa Sochat-->

<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="generator" content="Hugo 0.55.6" />

<META NAME="ROBOTS" CONTENT="INDEX, FOLLOW">

<link rel="alternate" type="application/rss&#43;xml" href="/docs/index.xml">

<link rel="shortcut icon" href="/singularityhub-archive/assets/favicons/favicon.ico" >
<link rel="apple-touch-icon" href="/singularityhub-archive/assets/favicons/apple-touch-icon-180x180.png" sizes="180x180">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/favicon-16x16.png" sizes="16x16">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/favicon-32x32.png" sizes="32x32">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/android-36x36.png" sizes="36x36">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/android-48x48.png" sizes="48x48">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/android-72x72.png" sizes="72x72">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/android-96x196.png" sizes="96x196">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/android-144x144.png" sizes="144x144">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/android-192x192.png"sizes="192x192">

<title>Thakk Biobase Latest</title>
<meta property="og:title" content="Thakk Biobase Latest" />
<meta property="og:description" content="This is the Singularity Hub archive, where you can search for (still pull-able) Singularity images and recipes from the original Singularity Hub, active from 2016 until 2021.
" />
<meta property="og:type" content="website" />
<meta property="og:url" content="https://singularityhub.github.io" />
<meta property="og:site_name" content="https://singularityhub.github.io" />

<meta itemprop="name" content="Thakk Biobase Latest">
<meta itemprop="description" content="This is the Singularity Hub archive, where you can search for (still pull-able) Singularity images and recipes from the original Singularity Hub, active from 2016 until 2021.
">

<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Thakk Biobase Latest"/>
<meta name="twitter:description" content="This is the Singularity Hub archive, where you can search for (still pull-able) Singularity images and recipes from the original Singularity Hub, active from 2016 until 2021.
"/>

<link rel="stylesheet" href="/singularityhub-archive/assets/css/main.css">
<link rel="stylesheet" href="/singularityhub-archive/assets/css/palette.css">
<script
  src="/singularityhub-archive/assets/js/jquery-3.3.1/jquery-3.3.1.min.js"
  integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8="
  crossorigin="anonymous"></script>
</head>

  

  <body class="td-section">
    <header>
        <nav class="js-navbar-scroll navbar navbar-expand navbar-dark flex-column flex-md-row td-navbar">
        <a class="navbar-brand" href="/singularityhub-archive/">
            <span class="navbar-logo"></span><img src="/singularityhub-archive/assets/img/logo.png" width="40px" style="padding-right:5px"><span class="text-uppercase font-weight-bold">Singularity Hub Archive</span>
	</a>
	<div class="td-navbar-nav-scroll ml-md-auto" id="main_navbar">
		<ul class="navbar-nav mt-2 mt-lg-0">
			<li class="nav-item mr-4 mb-2 mb-lg-0">
                            <a class="nav-link" href="https://github.com/singularityhub/singularityhub-archive" target="_blank"><span>GitHub</span></a>
			</li>
			<li class="nav-item mr-4 mb-2 mb-lg-0">
                            <a class="nav-link" href="/singularityhub-archive/about" ><span>About</span></a>
			</li>
			<li class="nav-item mr-4 mb-2 mb-lg-0">
                            <a class="nav-link" href="/singularityhub-archive/docs" ><span>Documentation</span></a>
			</li>
		</ul>
	</div>
	<div class="navbar-nav d-none d-lg-block">
 <input type="search" class="form-control td-search-input" placeholder="&#xf002 Search this site…" aria-label="Search this site…" autocomplete="off">
        </div>

	<div class="navbar-nav d-none d-lg-block">
          <a class="gh-source" data-gh-source="github" href="https://github.com/singularityhub/singularityhub-archive" title="Go to repository" data-md-state="done">
          <div class="gh-source__repository">
            <i class="fab fa fa-github fa-2x" style='padding-right:20px; float:left; margin-top:5px'></i>
            singularityhub/singularityhub-archive
          <ul class="gh-source__facts"><li class="gh-source__fact" id='stars'></li><li id="forks" class="gh-source__fact"></li></ul></div></a>
        </div>
      </div>


</nav>
</header>

<script>
$(document).ready(function() {
  var url = "https://api.github.com/search/repositories?q=singularityhub/singularityhub-archive";
  fetch(url, { 
      headers: {"Accept":"application/vnd.github.preview"}
  }).then(function(e) {
    return e.json()
  }).then(function(r) {
     console.log(r.items[0])
     stars = r.items[0]['stargazers_count']
     forks = r.items[0]['forks_count']
     $('#stars').text(stars + " Stars")
     $('#forks').text(forks + " Forks")
  });
});
</script>

    <div class="container-fluid td-outer">
      <div class="td-main">
        <div class="row flex-xl-nowrap">
          <div class="col-12 col-md-3 col-xl-2 td-sidebar d-print-none">
          <div id="td-sidebar-menu" class="td-sidebar__inner">
  
  <form class="td-sidebar__search d-flex align-items-center">
    <input type="search" class="form-control td-search-input" placeholder="&#xf002 Search this site…" aria-label="Search this site…" autocomplete="off">
      <button class="btn btn-link td-sidebar__toggle d-md-none p-0 ml-3 fas fa-bars" type="button" data-toggle="collapse" data-target="#td-section-nav" aria-controls="td-docs-nav" aria-expanded="false" aria-label="Toggle section navigation">
      </button>
  </form>  
  
  <nav class="collapse td-sidebar-nav pt-2 pl-4" id="td-section-nav">
    
      <ul class="td-sidebar-nav__section pr-md-3">
        
        
        
        <li class="td-sidebar-nav__section-title">
          <a  href="/singularityhub-archive/docs" class="align-left pl-0 pr-2 active td-sidebar-link td-sidebar-link__section">Documentation</a>
        </li>
        
          <ul>
            <li class="collapse show" id="documentation">
              <ul class="td-sidebar-nav__section pr-md-3">
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                /singularityhub-archive/
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">Collections
                    </a>
                  </li>
                  
                    <ul>
                      <li class="collapse show" id="">
                        
                      </li>
                    </ul>
                  
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                /singularityhub-archive/about
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">About
                    </a>
                  </li>
                  
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                /singularityhub-archive/news
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">News
                    </a>
                  </li>
                  
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                /singularityhub-archive/tags
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">Tags
                    </a>
                  </li>
                  
                
              </ul>
            </li>
          </ul>
        
      </ul>
    
      <ul class="td-sidebar-nav__section pr-md-3">
        
        
        
        <li class="td-sidebar-nav__section-title">
          <a  href="" class="align-left pl-0 pr-2 active td-sidebar-link td-sidebar-link__section">External</a>
        </li>
        
          <ul>
            <li class="collapse show" id="external">
              <ul class="td-sidebar-nav__section pr-md-3">
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                https://singularityhub.github.io/singularityhub-docs/
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">Singularity Hub Docs
                    </a>
                  </li>
                  
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                https://singularityhub.github.io/singularityhub-docs/lastday/
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">Singularity Hub Gallery
                    </a>
                  </li>
                  
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                http://datasets.datalad.org/?dir=/shub
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">Datalad Site
                    </a>
                  </li>
                  
                
              </ul>
            </li>
          </ul>
        
      </ul>
    
  </nav>
</div>

          </div>
          <div class="d-none d-xl-block col-xl-2 td-toc d-print-none">
              <div class="td-page-meta ml-2 pb-1 pt-2 mb-0">
                  <a href="https://github.com/singularityhub/singularityhub-archive/edit/main/_containers/thakk-biobase-latest.md" target="_blank"><i class="fa fa-edit fa-fw"></i> Edit this page</a>
<a href="https://github.com/singularityhub/singularityhub-archive/issues/new?labels=question&title=Question:&body=Question on: https://github.com/singularityhub/singularityhub-archive/tree/master/_containers/thakk-biobase-latest.md" target="_blank"><i class="fab fa-github fa-fw"></i> Create documentation issue</a>
<a href="https://github.com/singularityhub/singularityhub-archive/issues/new" target="_blank"><i class="fas fa-tasks fa-fw"></i> Create project issue</a>
<!-- this will parse through the header fields and add a button to open
     an issue / ask a question on Github. The editable field should be in
     the post frontend matter, and refer to the label to open the issue for -->

              </div>
              <nav id="TableOfContents"><ul>
              <li><ul id="TOC">
                <!-- Links will be appended here-->
              </ul></li>
              </ul></nav>
          </div>
          <main class="col-12 col-md-9 col-xl-8 pl-md-5" role="main">
            <nav aria-label="breadcrumb" class="d-none d-md-block d-print-none">
	      <ol class="breadcrumb spb-1">
              
              
                <li class="breadcrumb-item">
                  <a href="https://singularityhub.github.io/singularityhub-archive/containers/">Containers</a>
                </li>
                
                <li class="breadcrumb-item active" aria-current="page">
	          <a href="https://singularityhub.github.io/singularityhub-archive/containers/thakk-biobase-latest/">Thakk Biobase Latest</a>
                </li>
                
	      </ol>
           </nav>
           <div class="td-content">
	      <h1 id="thakkbiobaselatest">thakk/biobase:latest</h1>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>singularity pull shub://thakk/biobase:latest
</code></pre></div></div>

<h2 id="singularity-recipe">Singularity Recipe</h2>

<pre><code class="language-singularity">Bootstrap: docker
From: centos:8
Stage: build

%setup

%files

%help

   Bionformatics tools.

   Complete list of installed tools:
      - Samtools
      - BEDTools
      - FastQC
      - Bowtie2
      - MultiQC
      - Cutadapt
      - STAR
      - Hisat2
      - Picard
      - Trimmomatic
      - Samblaster
      - VarScan
      - Vcfanno
      - Plink
      - MACS2
      - Homer
      - NextFlow
      - nf-core
      - MAGeCK
      - TrimGalore
      - Bismark
      - Subread
      - UCSC tools:
         - addCols
         - ameme
         - autoDtd
         - autoSql
         - autoXml
         - ave
         - aveCols
         - axtChain
         - axtSort
         - axtSwap
         - axtToMaf
         - axtToPsl
         - bamToPsl
         - bedClip
         - bedCommonRegions
         - bedCoverage
         - bedExtendRanges
         - bedGeneParts
         - bedGraphPack
         - bedGraphToBigWig
         - bedIntersect
         - bedItemOverlapCount
         - bedJoinTabOffset
         - bedJoinTabOffset.py
         - bedMergeAdjacent
         - bedPartition
         - bedPileUps
         - bedRemoveOverlap
         - bedRestrictToPositions
         - bedSort
         - bedToBigBed
         - bedToExons
         - bedToGenePred
         - bedToPsl
         - bedWeedOverlapping
         - bigBedInfo
         - bigBedNamedItems
         - bigBedSummary
         - bigBedToBed
         - bigMafToMaf
         - bigPslToPsl
         - bigWigAverageOverBed
         - bigWigCat
         - bigWigCluster
         - bigWigCorrelate
         - bigWigInfo
         - bigWigMerge
         - bigWigSummary
         - bigWigToBedGraph
         - bigWigToWig
         - binFromRange
         - blastToPsl
         - blastXmlToPsl
         - blat
         - calc
         - catDir
         - catUncomment
         - chainAntiRepeat
         - chainBridge
         - chainFilter
         - chainMergeSort
         - chainNet
         - chainPreNet
         - chainSort
         - chainSplit
         - chainStitchId
         - chainSwap
         - chainToAxt
         - chainToPsl
         - chainToPslBasic
         - checkAgpAndFa
         - checkCoverageGaps
         - checkHgFindSpec
         - checkTableCoords
         - chopFaLines
         - chromGraphFromBin
         - chromGraphToBin
         - clusterGenes
         - colTransform
         - countChars
         - cpg_lh
         - crTreeIndexBed
         - crTreeSearchBed
         - dbSnoop
         - dbTrash
         - endsInLf
         - estOrient
         - expMatrixToBarchartBed
         - faAlign
         - faCmp
         - faCount
         - faFilter
         - faFilterN
         - faFrag
         - faNoise
         - faOneRecord
         - faPolyASizes
         - faRandomize
         - faRc
         - faSize
         - faSomeRecords
         - faSplit
         - faToFastq
         - faToTab
         - faToTwoBit
         - faTrans
         - fastqStatsAndSubsample
         - fastqToFa
         - featureBits
         - fetchChromSizes
         - findMotif
         - fixStepToBedGraph.pl
         - gapToLift
         - genePredCheck
         - genePredFilter
         - genePredHisto
         - genePredSingleCover
         - genePredToBed
         - genePredToBigGenePred
         - genePredToFakePsl
         - genePredToGtf
         - genePredToMafFrames
         - genePredToProt
         - gensub2
         - getRna
         - getRnaPred
         - gfClient
         - gfServer
         - gff3ToGenePred
         - gff3ToPsl
         - gmtime
         - gtfToGenePred
         - headRest
         - hgBbiDbLink
         - hgFakeAgp
         - hgFindSpec
         - hgGcPercent
         - hgGoldGapGl
         - hgLoadBed
         - hgLoadChain
         - hgLoadGap
         - hgLoadMaf
         - hgLoadMafSummary
         - hgLoadNet
         - hgLoadOut
         - hgLoadOutJoined
         - hgLoadSqlTab
         - hgLoadWiggle
         - hgSpeciesRna
         - hgTrackDb
         - hgWiggle
         - hgsqldump
         - hgvsToVcf
         - hicInfo
         - htmlCheck
         - hubCheck
         - hubClone
         - hubPublicCheck
         - ixIxx
         - lavToAxt
         - lavToPsl
         - ldHgGene
         - liftOver
         - liftOverMerge
         - liftUp
         - linesToRa
         - localtime
         - mafAddIRows
         - mafAddQRows
         - mafCoverage
         - mafFetch
         - mafFilter
         - mafFrag
         - mafFrags
         - mafGene
         - mafMeFirst
         - mafNoAlign
         - mafOrder
         - mafRanges
         - mafSpeciesList
         - mafSpeciesSubset
         - mafSplit
         - mafSplitPos
         - mafToAxt
         - mafToBigMaf
         - mafToPsl
         - mafToSnpBed
         - mafsInRegion
         - makeTableList
         - maskOutFa
         - mktime
         - mrnaToGene
         - netChainSubset
         - netClass
         - netFilter
         - netSplit
         - netSyntenic
         - netToAxt
         - netToBed
         - newProg
         - newPythonProg
         - nibFrag
         - nibSize
         - oligoMatch
         - overlapSelect
         - para
         - paraFetch
         - paraHub
         - paraHubStop
         - paraNode
         - paraNodeStart
         - paraNodeStatus
         - paraNodeStop
         - paraSync
         - paraTestJob
         - parasol
         - positionalTblCheck
         - pslCDnaFilter
         - pslCat
         - pslCheck
         - pslDropOverlap
         - pslFilter
         - pslHisto
         - pslLiftSubrangeBlat
         - pslMap
         - pslMapPostChain
         - pslMrnaCover
         - pslPairs
         - pslPartition
         - pslPosTarget
         - pslPretty
         - pslRc
         - pslRecalcMatch
         - pslReps
         - pslScore
         - pslSelect
         - pslSomeRecords
         - pslSort
         - pslStats
         - pslSwap
         - pslToBed
         - pslToBigPsl
         - pslToChain
         - pslToPslx
         - pslxToFa
         - qaToQac
         - qacAgpLift
         - qacToQa
         - qacToWig
         - raSqlQuery
         - raToLines
         - raToTab
         - randomLines
         - rmFaDups
         - rowsToCols
         - sizeof
         - spacedToTab
         - splitFile
         - splitFileByColumn
         - sqlToXml
         - strexCalc
         - stringify
         - subChar
         - subColumn
         - tailLines
         - tdbQuery
         - tdbRename
         - tdbSort
         - textHistogram
         - tickToDate
         - toLower
         - toUpper
         - transMapPslToGenePred
         - trfBig
         - twoBitDup
         - twoBitInfo
         - twoBitMask
         - twoBitToFa
         - ucscApiClient
         - vai.pl
         - validateFiles
         - validateManifest
         - varStepToBedGraph.pl
         - webSync
         - wigCorrelate
         - wigEncode
         - wigToBigWig
         - wordLine
         - xmlCat
         - xmlToSql


## END HELP



%post

   NOW=`date`
   echo "export NOW=\"${NOW}\"" &gt;&gt; $SINGULARITY_ENVIRONMENT
   set -e

   dnf -y makecache
   dnf -y group install "Development Tools"
   dnf -y install --allowerasing  hostname which dnf-utils \
     python3-pip git zlib zlib-devel bzip2 bzip2-devel \
     xz xz-devel libcurl libcurl-devel ncurses ncurses-devel \
     unzip wget gnuplot rsync java-1.8.0-openjdk java-1.8.0-openjdk-devel \
     openssl-devel libffi-devel python36-devel

   ln -s /usr/bin/python3 /usr/bin/python
   ln -s /usr/bin/pip3 /usr/bin/pip

   pip install numpy pandas matplotlib scikit-learn scipy opencv-contrib-python

   # Intel Threading Building Blocks 2019 U8
   echo "/usr/local/lib" &gt;&gt; /etc/ld.so.conf.d/usr_local_lib.conf
   rm -rf tbb
   git clone https://github.com/intel/tbb
   cd tbb
   make -j 8
   find . -name '*so' | grep release | xargs -I{} cp {} /usr/local/lib/  # Copy release libs to sys dir
   find . -name '*so.2' | grep release | xargs -I{} cp {} /usr/local/lib/
   cp -R include/* /usr/local/include/
   cd ..
   ldconfig
   mkdir -p /usr/local/lib64/python3.6/site-packages


%runscript

    echo "Container was created $NOW"
    echo "Arguments received: $*"
    exec echo "$@"


%labels
   Author "tomi.hakkinen@tuni.fi and francesco.tabaro@tuni.fi"
   Version 0.29

%environment
#    GLOBAL=variables
#    AVAILABLE="to all apps"
   export PATH=$PATH:/usr/bin:/usr/local/bin/homer/bin:/opt/ucsc_tools:/opt/Bismark-0.22.3
   LANG="en_US.UTF-8"
   export LANG
   PYTHONWARNINGS=ignore::yaml.YAMLLoadWarning

# ################### Samtools #################################

%apprun samtools
   exec samtools

%appinstall samtools
   rm -rf htslib
   git clone https://github.com/samtools/htslib
   cd htslib
   autoheader
   autoconf
   ./configure
   make -j 8
   make install
   cd ..
   rm -rf samtools
   git clone https://github.com/samtools/samtools
   cd samtools
   autoheader
   autoconf -Wno-syntax
   ./configure
   make -j 8
   make install
   cd ..

%apphelp samtools
   Samtools 1.9. https://github.com/samtools/samtools

# ######################## Bedtools ###################################

%apprun bedtools
    exec bedtools

%appinstall bedtools
   rm -rf bedtools2
   git clone https://github.com/arq5x/bedtools2/
   #wget https://github.com/arq5x/bedtools2/releases/download/v2.28.0/
   #bedtools-2.28.0.tar.gz
   #tar -zxf bedtools-2.28.0.tar.gz
   cd bedtools2
   make -j 8
   cp bin/* /usr/local/bin/

%apphelp bedtools
    Bedtools 2.28.0. https://github.com/arq5x/bedtools2/


# ######################## FastQC ###################################

%apprun fastqc
   exec fastqc

%appinstall fastqc
   #git clone https://github.com/s-andrews/FastQC
   cd /usr/local/bin
   wget http://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.8.zip
   unzip -o fastqc_v0.11.8.zip
   chmod +x FastQC/fastqc
   ln -sf FastQC/fastqc .
   rm fastqc_v0.11.8.zip

%apphelp fastqc
   FastQC v0.11.8. https://github.com/s-andrews/FastQC


# ######################## Bowtie2 ################################

%apprun bowtie2
   exec bowtie2

%appinstall bowtie2
   wget -O bowtie.zip https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.3.5.1/bowtie2-2.3.5.1-source.zip/download
   unzip -o bowtie.zip
   cd bowtie2-2.3.5.1
   make -j 8
   cp bowtie2* /usr/local/bin/
   cd ..


%apphelp bowtie2
   Bowtie 2 v. 2.3.5.1 . http://bowtie-bio.sourceforge.net/bowtie2/index.shtml


# ######################## MultiQC ##################################

%apprun multiqc
   exec multiqc

%appinstall multiqc
   pip install  git+https://github.com/ewels/MultiQC.git

%apphelp multiqc
   MultiQC v1.8dev. https://multiqc.info/

# ######################## Cutadapt ################################

%apprun cutadapt
   exec cutadapt

%appinstall cutadapt
   pip install cutadapt

%apphelp cutadapt
   Cutadapt 2.4.. https://cutadapt.readthedocs.io/en/stable/guide.html

# ######################### STAR ###################################

%apprun star
   exec star

%appinstall star
   rm -rf STAR
   git clone https://github.com/alexdobin/STAR.git
   cd STAR/source
   make -j 8 STAR
   cp STAR /usr/local/bin/
   ln -sf /usr/local/bin/STAR /usr/local/bin/star
   cd ..
   cd ..

%apphelp star
   STAR 2.7.1a. https://github.com/alexdobin/STAR

# ######################## HISAT2 #################################

%apprun hisat2
   exec hisat2

%appinstall hisat2
   rm -rf hisat2-2.1.0
   wget http://ccb.jhu.edu/software/hisat2/dl/hisat2-2.1.0-source.zip
   unzip hisat2-2.1.0-source.zip
   cd hisat2-2.1.0
   make -j 8
   #find . -name '*py' | xargs -I{} cp {} /usr/local/bin/
   find . -executable -type f | grep -v tinythread | xargs -I{} cp {} /usr/local/bin/
   cd ..

%apphelp hisat2
   HISAT2 2.1.0. https://ccb.jhu.edu/software/hisat2/index.shtml


# ######################## Picard #########################

%apprun picard
   exec picard

%appinstall picard
   wget https://github.com/broadinstitute/picard/releases/download/2.20.5/picard.jar
   mv picard.jar /usr/local/bin
   echo "#"'!'"/usr/bin/bash" &gt; /usr/local/bin/picard
   echo "java -jar /usr/local/bin/picard.jar \"\$@\"" &gt;&gt; /usr/local/bin/picard
   chmod +x /usr/local/bin/picard

%apphelp picard
   Picard 2.20.5. http://broadinstitute.github.io/picard/


# ######################### Trimmomatic #######################

%apprun trimmomatic
   exec trimmomatic

%appinstall trimmomatic
   wget http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.39.zip
   unzip -o Trimmomatic-0.39.zip -d /usr/local/bin
   echo "#"'!'"/usr/bin/bash" &gt; /usr/local/bin/trimmomatic
   echo "java -jar /usr/local/bin/Trimmomatic-0.39/trimmomatic-0.39.jar \"\$@\"" &gt;&gt; /usr/local/bin/trimmomatic
   chmod +x /usr/local/bin/trimmomatic
   rm Trimmomatic-0.39.zip

%apphelp trimmomatic
   Trimmomatic 0.39. http://www.usadellab.org/cms/?page=trimmomatic


# ######################### Samblaster ############################

%apprun samblaster
   exec samblaster

%appinstall samblaster
   git clone git://github.com/GregoryFaust/samblaster.git
   cd samblaster
   make -j 8
   cp samblaster /usr/local/bin/
   cd ..
   rm -r samblaster

%apphelp samblaster
   Samblaster 0.1.24 . https://github.com/GregoryFaust/samblaster


# ########################## VarScan #################################

%apprun varscan
   exec varscan

%appinstall varscan
   wget https://github.com/dkoboldt/varscan/raw/master/VarScan.v2.4.4.jar
   mv VarScan.v2.4.4.jar /usr/local/bin/
   echo "#"'!'"/usr/bin/bash" &gt; /usr/local/bin/varscan
   echo "java -jar /usr/local/bin/VarScan.v2.4.4.jar \"\$@\"" &gt;&gt; /usr/local/bin/varscan
   chmod +x /usr/local/bin/varscan

%apphelp varscan
   VarScan 2.4.4 . https://github.com/dkoboldt/varscan


# ########################### VcfAnno ###############################

%apprun vcfanno
   exec vcfanno

%appinstall vcfanno
   wget https://github.com/brentp/vcfanno/releases/download/v0.3.2/vcfanno_linux64 -O /usr/local/bin/vcfanno
   chmod +x /usr/local/bin/vcfanno

%apphelp vcfanno
   VcfAnno 0.3.2 . https://github.com/brentp/vcfanno/releases/


# ############################ Plink ##################################

%apprun plink
   exec plink

%appinstall plink
   wget http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190617.zip
   unzip plink_linux_x86_64_20190617.zip
   mv plink /usr/local/bin/
   mv prettify /usr/local/bin/


%apphelp plink
   PLINK v1.90b6.10 64-bit (17 Jun 2019) . https://www.cog-genomics.org/plink2


# ############################ MACS2 ######################################

%apprun macs2
   exec macs2

%appinstall macs2
   export PYTHONPATH=$PYTHONPATH:/usr/local/lib64/python3.6/site-packages/
   git clone https://github.com/taoliu/MACS.git
   cd MACS/
   python setup.py install --prefix /usr/local
   cd ..
   rm -r MACS

%apphelp macs2
   MACS2 2.2.4 ( Oct 7 2019) . https://github.com/taoliu/MACS

# ############################ Homer ########################

%apprun homer
   exec homer

%appinstall homer
   mkdir -p /usr/local/bin/homer
   cd /usr/local/bin/homer
   wget http://homer.ucsd.edu/homer/configureHomer.pl
   perl configureHomer.pl -install

%apphelp homer
   Homer 4.10 ( Oct 7 2019 ) . http://homer.ucsd.edu/homer/

# ############################ Nextflow #######################

%apprun nextflow
   exec nextflow

%appinstall nextflow
   cd /usr/local/bin
   curl -s https://get.nextflow.io | bash
   chmod 777 /usr/local/bin/nextflow

%apphelp nextflow
   Nextflow 19.09 ( Oct 7 2019 ) . https://www.nextflow.io/index.html

# ############################ nf-core ########################

%apprun nfcore
   exec nf-core

%appinstall nfcore
   LANG="en_US.UTF-8"
   export LANG
   pip install --upgrade --force-reinstall git+https://github.com/nf-core/tools.git

%apphelp nfcore
   Nf-Core . https://nf-co.re/ . A community effort to collect a curated set of analysis pipelines built using Nextflow.

# ############################ MAGeCK ########################

%apprun mageck
   exec mageck

%appinstall mageck
   wget "https://downloads.sourceforge.net/project/mageck/0.5/liulab-mageck-0.5.9.2.tar.gz"
   tar zxvf liulab-mageck-0.5.9.2.tar.gz
   cd liulab-mageck-ef7c39474ed0
   python setup.py install
   cd ..
   rm -r liulab-mageck-ef7c39474ed0

%apphelp mageck
   MAGeCK (0.5.9.2) enables robust identification of essential genes from genome-scale CRISPR/Cas9 knockout screens

# ############################ TrimGalore ########################

%apprun trim_galore
   exec trim_galore

%appinstall trim_galore
   curl -fsSL https://github.com/FelixKrueger/TrimGalore/archive/0.6.5.tar.gz -o trim_galore.tar.gz
    tar xzf trim_galore.tar.gz
    cp TrimGalore-0.6.5/trim_galore /usr/local/bin
    rm -r TrimGalore-0.6.5


%apphelp trim_galore
    TrimGalore (0.6.5) . https://github.com/FelixKrueger/TrimGalore


# ############################ Bismark ########################

%apprun bismark
   exec bismark

%appinstall bismark
    curl -fsSL https://github.com/FelixKrueger/Bismark/archive/0.22.3.tar.gz -o Bismark_0.22.3.tar.gz
    tar xzf Bismark_0.22.3.tar.gz -C /opt/

%apphelp bismark
    Bismark (0.22.3) . https://github.com/FelixKrueger/Bismark


# ############################ Subread package ###################

%apprun featureCounts
	exec featureCounts

%appinstall featureCounts
	wget -O subread.tar.gz https://sourceforge.net/projects/subread/files/subread-2.0.1/subread-2.0.1-source.tar.gz/download
	tar zxvf subread.tar.gz
	cd subread-2.0.1-source/src
	make -f Makefile.Linux
	cd ../bin
	cp -R * /usr/local/bin/
	cd ../..
	rm subread.tar.gz

%apphelp featureCounts
	Subread 2.0.1 featureCounts http://bioinf.wehi.edu.au/subread-package/

# ############################ UCSC tools ########################

%appinstall ucsc_tools
    mkdir -p /opt/ucsc_tools
    cd /opt/ucsc_tools
    rsync -aP rsync://hgdownload.soe.ucsc.edu/genome/admin/exe/linux.x86_64/ ./


%apphelp ucsc_tools
    UCSC tools (v390) . http://hgdownload.soe.ucsc.edu/admin/exe/ DOC: http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/FOOTER.txt
</code></pre>

<h2 id="collection">Collection</h2>

<ul>
  <li>Name: <a href="https://github.com/thakk/biobase">thakk/biobase</a></li>
  <li>License: None</li>
</ul>




<br>
<a class="btn btn-primary" href="https://datasets.datalad.org?dir=/shub/thakk/biobase/latest/2021-02-27-d76758db-10d20187/" target="_blank" type="button">View on Datalad</a>

<h2>Metrics</h2>

<table>
<tr>
  <th>key</th>
  <th>value</th>
</tr>
<tbody>
  <tr>
    <td>id</td>
    <td>/containers/thakk-biobase-latest</td>
  </tr>
  <tr>
    <td>collection name</td>
    <td>thakk/biobase</td>
  </tr>
  <tr>
    <td>branch</td>
    <td>master</td>
  </tr>
  <tr>
    <td>tag</td>
    <td>latest</td>
  </tr>
  <tr>
    <td>commit</td>
    <td>d76758db6efc723d136494cc0e0027bc680e808e</td>
  </tr>
  <tr>
    <td>version (container hash)</td>
    <td>10d20187508010e1ed85d13f0af77253</td>
  </tr>
  <tr>
    <td>build date</td>
    <td>2021-02-27T20:46:58.765Z</td>
  </tr>
  <tr>
    <td>size (MB)</td>
    <td>6474.0</td>
  </tr>
  <tr>
    <td>size (bytes)</td>
    <td>2668269599</td>
  </tr>
  <tr>
    <td>SIF</td>
    <td><a href="https://datasets.datalad.org/shub/thakk/biobase/latest/2021-02-27-d76758db-10d20187/10d20187508010e1ed85d13f0af77253.sif">Download URL (please use pull with shub://)</td>
  </tr>
  <tr>
    <td>Datalad URL</td>
    <td><a href="https://datasets.datalad.org?dir=/shub/thakk/biobase/latest/2021-02-27-d76758db-10d20187/">View on Datalad</td>
  </tr>
  <tr>
    <td>Singularity Recipe</td>
    <td><a href="https://datasets.datalad.org/shub/thakk/biobase/latest/2021-02-27-d76758db-10d20187/Singularity">Singularity Recipe on Datalad</td>
  </tr>
</tbody>
</table>
  
<small>We cannot guarantee that all containers will still exist on GitHub.</small>

<script
  src="/singularityhub-archive/assets/js/jquery-3.3.1/jquery-3.3.1.min.js"
  integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8="
  crossorigin="anonymous"></script>

<script>
$(document).ready(function() {

    var toc = $('#TOC');

    // Select each header
    sections = $('.td-content h1');
        $.each(sections, function(idx, v) {
            section = $(v);
            var div_id = $(section).attr('id');
            var div_text = section.text().split('¶')[0];
            var parent = $("#" + div_id)
            var content = '<li id="link_' + div_id + '" class="md-nav__item"><a class="md-nav__link" href="#' + div_id + '" title="' + div_text +'">' + div_text +'</a></li>';
            $(toc).append(content);

            // Add section code to subnavigation
            var children = $('<nav class="md-nav"><ul class="md-nav__list"></nav></ul>')
            var contenders = $("#" + div_id).nextUntil("h1");
            $.each(contenders, function(idx, contender){
               if($(contender).is('h2') || $(contender).is('h3')) {
                   var contender_id = $(contender).attr('id');
                   var contender_text = $(contender).text().split('¶')[0];
                   var content = '<li class="md-nav__item"><a class="md-nav__link" href="#' + contender_id + '" title="' + contender_text +'">' + contender_text +'</a></li>';
                   children.append(content);
                }
             })
             $("#link_" + div_id).append(children);
        });
    });
</script>

<script>
var headers = ["h1", "h2", "h3", "h4"]
var colors = ["red", "orange", "green", "blue"]

$.each(headers, function(i, header){
    var color = colors[i];
    $(header).each(function () {
        var href=$(this).attr("id");
        $(this).append('<a class="headerlink" style="color:' + color + '" href="#' + href + '" title="Permanent link">¶</a>')
    });
})
</script>

<script>
$('h1').first().append('<div><span style="font-size:12px" class="badge badge-danger"><a style="cursor:pointer; color:white" href="https://singularityhub.github.io/singularityhub-archive/tags#latest ">latest</a></span></div>')</script>

	
              
              <style>
  .feedback--answer {
    display: inline-block;
  }
  .feedback--answer-no {
    margin-left: 1em;
  }
  .feedback--response {
    display: none;
    margin-top: 1em;
  }
  .feedback--response__visible {
    display: block;
  }
</style>
<h5 class="feedback--title">Feedback</h5>
<p class="feedback--question">Was this page helpful?</p>
<button class="feedback--answer feedback--answer-yes">Yes</button>
<button class="feedback--answer feedback--answer-no">No</button>
<p class="feedback--response feedback--response-yes">
  Glad to hear it! Please <a href="https://github.com/singularityhub/singularityhub-archive/issues/new">tell us how we can improve</a>.
</p>
<p class="feedback--response feedback--response-no">
  Sorry to hear that. Please <a href="https://github.com/singularityhub/singularityhub-archive/issues/new">tell us how we can improve</a>.
</p>
<script>
  const yesButton = document.querySelector('.feedback--answer-yes');
  const noButton = document.querySelector('.feedback--answer-no');
  const yesResponse = document.querySelector('.feedback--response-yes');
  const noResponse = document.querySelector('.feedback--response-no');
  const disableButtons = () => {
    yesButton.disabled = true;
    noButton.disabled = true;
  };
  const sendFeedback = (value) => {
    if (typeof ga !== 'function') return;
    const args = {
      command: 'send',
      hitType: 'event',
      category: 'Helpful',
      action: 'click',
      label: window.location.pathname,
      value: value
    };
    ga(args.command, args.hitType, args.category, args.action, args.label, args.value);
  };
  yesButton.addEventListener('click', () => {
    yesResponse.classList.add('feedback--response__visible');
    disableButtons();
    sendFeedback(1);
  });
  noButton.addEventListener('click', () => {
    noResponse.classList.add('feedback--response__visible');
    disableButtons();
    sendFeedback(0);
  });
</script><br/>


           </div>
          </main>
        </div>
      </div>
      <footer class="bg-dark py-5 row d-print-none">
  <div class="container-fluid mx-sm-5">
    <div class="row">
      <div class="col-6 col-sm-4 text-xs-center order-sm-2">
        <ul class="list-inline mb-0">  
          
            <li class="list-inline-item mx-2 h3" data-toggle="tooltip" data-placement="top" title="" aria-label="Twitter" data-original-title="Twitter">
              <a class="text-white" target="_blank" href="https://twitter.com/vsoch">
                <i class="fab fa-twitter"></i>
              </a>
            </li>
          
          
            <li class="list-inline-item mx-2 h3" data-toggle="tooltip" data-placement="top" title="" aria-label="LinkedIn" data-original-title="LinkedIn">
              <a class="text-white" target="_blank" href="https://linkedin.com/in/vsochat">
                <i class="fab fa-linkedin"></i>
              </a>
            </li>
          
        </ul>
      </div>
      <div class="col-6 col-sm-4 text-right text-xs-center order-sm-3">
        <ul class="list-inline mb-0">  
          <li class="list-inline-item mx-2 h3" data-toggle="tooltip" data-placement="top" title="" aria-label="GitHub" data-original-title="GitHub">
            <a class="text-white" target="_blank" href="https://github.com/singularityhub/singularityhub-archive">
              <i class="fab fa-github"></i>
            </a>
          </li>
        </ul>
      </div>
      <div class="col-12 col-sm-4 text-center py-2 order-sm-2">
        <small class="text-white">© 2021 Dinosaur Avocado All Rights Reserved</small>
        
        <p class="mt-2"><a href="/about/">About the Archive</a></p>	
      </div>
    </div>
  </div>
</footer>

    </div>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js" integrity="sha384-ZMP7rVo3mIykV+2+9J3UJ46jBk0WLaUAdn689aCwoqbBJiSnjAK/l8WvCWPIPm49" crossorigin="anonymous"></script>
<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js" integrity="sha384-ChfqqxuZUCnJSK3+MXmPNIyE6ZbWh2IMqE241rYiqJxyMiZ6OW/JmZQ5stwEULTy" crossorigin="anonymous"></script>
<script src="/singularityhub-archive/assets/js/main.js"></script>

  </body>
</html>
