<!DOCTYPE html>
<html lang="en" class="no-js">
  <!-- Copyright 2019-2021 Vanessa Sochat-->
  <head>
<meta charset="utf-8">
<!-- Copyright 2019-2021 Vanessa Sochat-->

<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="generator" content="Hugo 0.55.6" />

<META NAME="ROBOTS" CONTENT="INDEX, FOLLOW">

<link rel="alternate" type="application/rss&#43;xml" href="/docs/index.xml">

<link rel="shortcut icon" href="/singularityhub-archive/assets/favicons/favicon.ico" >
<link rel="apple-touch-icon" href="/singularityhub-archive/assets/favicons/apple-touch-icon-180x180.png" sizes="180x180">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/favicon-16x16.png" sizes="16x16">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/favicon-32x32.png" sizes="32x32">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/android-36x36.png" sizes="36x36">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/android-48x48.png" sizes="48x48">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/android-72x72.png" sizes="72x72">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/android-96x196.png" sizes="96x196">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/android-144x144.png" sizes="144x144">
<link rel="icon" type="image/png" href="/singularityhub-archive/assets/favicons/android-192x192.png"sizes="192x192">

<title>Scinethpc Singularity Crisprcasfinder Latest</title>
<meta property="og:title" content="Scinethpc Singularity Crisprcasfinder Latest" />
<meta property="og:description" content="This is the Singularity Hub archive, where you can search for (still pull-able) Singularity images and recipes from the original Singularity Hub, active from 2016 until 2021.
" />
<meta property="og:type" content="website" />
<meta property="og:url" content="https://singularityhub.github.io" />
<meta property="og:site_name" content="https://singularityhub.github.io" />

<meta itemprop="name" content="Scinethpc Singularity Crisprcasfinder Latest">
<meta itemprop="description" content="This is the Singularity Hub archive, where you can search for (still pull-able) Singularity images and recipes from the original Singularity Hub, active from 2016 until 2021.
">

<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Scinethpc Singularity Crisprcasfinder Latest"/>
<meta name="twitter:description" content="This is the Singularity Hub archive, where you can search for (still pull-able) Singularity images and recipes from the original Singularity Hub, active from 2016 until 2021.
"/>

<link rel="stylesheet" href="/singularityhub-archive/assets/css/main.css">
<link rel="stylesheet" href="/singularityhub-archive/assets/css/palette.css">
<script
  src="/singularityhub-archive/assets/js/jquery-3.3.1/jquery-3.3.1.min.js"
  integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8="
  crossorigin="anonymous"></script>
</head>

  

  <body class="td-section">
    <header>
        <nav class="js-navbar-scroll navbar navbar-expand navbar-dark flex-column flex-md-row td-navbar">
        <a class="navbar-brand" href="/singularityhub-archive/">
            <span class="navbar-logo"></span><img src="/singularityhub-archive/assets/img/logo.png" width="40px" style="padding-right:5px"><span class="text-uppercase font-weight-bold">Singularity Hub Archive</span>
	</a>
	<div class="td-navbar-nav-scroll ml-md-auto" id="main_navbar">
		<ul class="navbar-nav mt-2 mt-lg-0">
			<li class="nav-item mr-4 mb-2 mb-lg-0">
                            <a class="nav-link" href="https://github.com/singularityhub/singularityhub-archive" target="_blank"><span>GitHub</span></a>
			</li>
			<li class="nav-item mr-4 mb-2 mb-lg-0">
                            <a class="nav-link" href="/singularityhub-archive/about" ><span>About</span></a>
			</li>
			<li class="nav-item mr-4 mb-2 mb-lg-0">
                            <a class="nav-link" href="/singularityhub-archive/docs" ><span>Documentation</span></a>
			</li>
		</ul>
	</div>
	<div class="navbar-nav d-none d-lg-block">
 <input type="search" class="form-control td-search-input" placeholder="&#xf002 Search this site…" aria-label="Search this site…" autocomplete="off">
        </div>

	<div class="navbar-nav d-none d-lg-block">
          <a class="gh-source" data-gh-source="github" href="https://github.com/singularityhub/singularityhub-archive" title="Go to repository" data-md-state="done">
          <div class="gh-source__repository">
            <i class="fab fa fa-github fa-2x" style='padding-right:20px; float:left; margin-top:5px'></i>
            singularityhub/singularityhub-archive
          <ul class="gh-source__facts"><li class="gh-source__fact" id='stars'></li><li id="forks" class="gh-source__fact"></li></ul></div></a>
        </div>
      </div>


</nav>
</header>

<script>
$(document).ready(function() {
  var url = "https://api.github.com/search/repositories?q=singularityhub/singularityhub-archive";
  fetch(url, { 
      headers: {"Accept":"application/vnd.github.preview"}
  }).then(function(e) {
    return e.json()
  }).then(function(r) {
     console.log(r.items[0])
     stars = r.items[0]['stargazers_count']
     forks = r.items[0]['forks_count']
     $('#stars').text(stars + " Stars")
     $('#forks').text(forks + " Forks")
  });
});
</script>

    <div class="container-fluid td-outer">
      <div class="td-main">
        <div class="row flex-xl-nowrap">
          <div class="col-12 col-md-3 col-xl-2 td-sidebar d-print-none">
          <div id="td-sidebar-menu" class="td-sidebar__inner">
  
  <form class="td-sidebar__search d-flex align-items-center">
    <input type="search" class="form-control td-search-input" placeholder="&#xf002 Search this site…" aria-label="Search this site…" autocomplete="off">
      <button class="btn btn-link td-sidebar__toggle d-md-none p-0 ml-3 fas fa-bars" type="button" data-toggle="collapse" data-target="#td-section-nav" aria-controls="td-docs-nav" aria-expanded="false" aria-label="Toggle section navigation">
      </button>
  </form>  
  
  <nav class="collapse td-sidebar-nav pt-2 pl-4" id="td-section-nav">
    
      <ul class="td-sidebar-nav__section pr-md-3">
        
        
        
        <li class="td-sidebar-nav__section-title">
          <a  href="/singularityhub-archive/docs" class="align-left pl-0 pr-2 active td-sidebar-link td-sidebar-link__section">Documentation</a>
        </li>
        
          <ul>
            <li class="collapse show" id="documentation">
              <ul class="td-sidebar-nav__section pr-md-3">
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                /singularityhub-archive/
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">Collections
                    </a>
                  </li>
                  
                    <ul>
                      <li class="collapse show" id="">
                        
                      </li>
                    </ul>
                  
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                /singularityhub-archive/about
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">About
                    </a>
                  </li>
                  
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                /singularityhub-archive/news
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">News
                    </a>
                  </li>
                  
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                /singularityhub-archive/tags
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">Tags
                    </a>
                  </li>
                  
                
              </ul>
            </li>
          </ul>
        
      </ul>
    
      <ul class="td-sidebar-nav__section pr-md-3">
        
        
        
        <li class="td-sidebar-nav__section-title">
          <a  href="" class="align-left pl-0 pr-2 active td-sidebar-link td-sidebar-link__section">External</a>
        </li>
        
          <ul>
            <li class="collapse show" id="external">
              <ul class="td-sidebar-nav__section pr-md-3">
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                https://singularityhub.github.io/singularityhub-docs/
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">Singularity Hub Docs
                    </a>
                  </li>
                  
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                https://singularityhub.github.io/singularityhub-docs/lastday/
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">Singularity Hub Gallery
                    </a>
                  </li>
                  
                
                  <li class="td-sidebar-nav__section-title">
                    <a href="
                                http://datasets.datalad.org/?dir=/shub
                             " 
                       class="align-left pl-0 pr-2 td-sidebar-link td-sidebar-link__section">Datalad Site
                    </a>
                  </li>
                  
                
              </ul>
            </li>
          </ul>
        
      </ul>
    
  </nav>
</div>

          </div>
          <div class="d-none d-xl-block col-xl-2 td-toc d-print-none">
              <div class="td-page-meta ml-2 pb-1 pt-2 mb-0">
                  <a href="https://github.com/singularityhub/singularityhub-archive/edit/main/_containers/SciNetHPC-singularity-CRISPRCasFinder-latest.md" target="_blank"><i class="fa fa-edit fa-fw"></i> Edit this page</a>
<a href="https://github.com/singularityhub/singularityhub-archive/issues/new?labels=question&title=Question:&body=Question on: https://github.com/singularityhub/singularityhub-archive/tree/master/_containers/SciNetHPC-singularity-CRISPRCasFinder-latest.md" target="_blank"><i class="fab fa-github fa-fw"></i> Create documentation issue</a>
<a href="https://github.com/singularityhub/singularityhub-archive/issues/new" target="_blank"><i class="fas fa-tasks fa-fw"></i> Create project issue</a>
<!-- this will parse through the header fields and add a button to open
     an issue / ask a question on Github. The editable field should be in
     the post frontend matter, and refer to the label to open the issue for -->

              </div>
              <nav id="TableOfContents"><ul>
              <li><ul id="TOC">
                <!-- Links will be appended here-->
              </ul></li>
              </ul></nav>
          </div>
          <main class="col-12 col-md-9 col-xl-8 pl-md-5" role="main">
            <nav aria-label="breadcrumb" class="d-none d-md-block d-print-none">
	      <ol class="breadcrumb spb-1">
              
              
                <li class="breadcrumb-item">
                  <a href="https://singularityhub.github.io/singularityhub-archive/containers/">Containers</a>
                </li>
                
                <li class="breadcrumb-item active" aria-current="page">
	          <a href="https://singularityhub.github.io/singularityhub-archive/containers/SciNetHPC-singularity-CRISPRCasFinder-latest/">Scinethpc Singularity Crisprcasfinder Latest</a>
                </li>
                
	      </ol>
           </nav>
           <div class="td-content">
	      <h1 id="scinethpcsingularity-crisprcasfinderlatest">SciNetHPC/singularity-CRISPRCasFinder:latest</h1>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>singularity pull shub://SciNetHPC/singularity-CRISPRCasFinder:latest
</code></pre></div></div>

<h2 id="singularity-recipe">Singularity Recipe</h2>

<pre><code class="language-singularity"># To build the container (Singularity version &gt;=2.6):
# sudo singularity build CRISPRCasFinder Singularity

BootStrap: docker
from: ubuntu:xenial

%labels
	MAINTAINER Bertrand Neron &lt;bneron@pasteur.fr&gt;
	AUTHOR Couvin David, Bernheim Aude, Toffano-Nioche Claire, Touchon Marie, Michalik Juraj, Neron Bertrand, Rocha Eduardo, Vergnaud Gilles, Gautheret Daniel, Pourcel Christine.
    CRISPRCasFinder.version 4.2.17

%help
    Name:
      CRISPRCasFinder standalone version 4.2.17

    Synopsis:
      A perl script to identify CRISPR arrays and associated Cas genes in DNA sequences

    Usage:
      ./CRISPRCasFinder.img &lt;filename.fasta&gt;
      OR
      singularity run CRISPRCasFinder.img [options] -in &lt;filename.fasta&gt;

      --Please note &lt;filename.fasta&gt; must be in Fasta format. Please also note that when several options are called, the option "-in or -i" must precede the input FASTA file.

    General:
      -help or -h           This help
      -version or -v        The current version of the program will be displayed

    Other options:

      [Input/Output and -so]
      -in or -i [XXX]       Input Fasta file (with extensions: .fasta, .fna, .mfa, .fa)
      -outdir or -out [XXX] Output directory (if users do not use this option, a delault directory will be created wit the date and time)
      -keepAll or -keep     Option allowing to keep secondary folders/files (Prodigal/Prokka, CasFinder, rawFASTA, Properties); (default: 0)
      -LOG or -log          Option allowing to write LOG files (default: 0)
      -HTML or -html        Option allowing to display results as a static HTML web page (default value: 0). The web page created (index.html) will be dependent of a CSS file (crispr.css)
      -copyCSS [XXX]        Option allowing to copy provided CSS file into "Visualization" repository if option -HTML is set (default: '/usr/local/share/CRISPRCasFinder/crispr.css')
      -soFile or -so [XXX]  Option allowing to use the shared object file if it is not present in current directory (default: '/.singularity.d/libs/sel392v2.so')

     [Detection of CRISPR arrays]
      -mismDRs or -md [XXX] Percentage mismatchs allowed between DRs (default: 20)
      -truncDR or -t [XXX]  Percentage mismatchs allowed for truncated DR (default: 33.3)
      -minDR or -mr [XXX]   Minimal size of DRs (default: 23)
      -maxDR or -xr [XXX]   Maximal size of DRs (default: 55)
      -minSP or -ms [XXX]   Minimal size of Spacers (default: 25)
      -maxSP or -xs [XXX]   Maximal size of Spacers (default: 60)
      -noMism or -n Option used to do not allow mismatches (default value is 1 when this option is not called. i.e. mismatches are allowed by default)
      -percSPmin or -pm [XXX]       Minimal Spacers size in function of DR size (default: 0.6)
      -percSPmax or -px [XXX]       Maximal Spacers size in function of DR size (default: 2.5)
      -spSim or -s [XXX]    Maximal allowed percentage of similarity between Spacers (default: 60)
      -DBcrispr or -dbc [XXX]       Option allowing to use a CSV file of all CRISPR candidates contained in CRISPRdb (from last update) (default: '/usr/local/share/CRISPRCasFinder/CRISPR_crisprdb.csv')
      -repeats or -rpts [XXX]       Option allowing to use a consensus repeats list generated by CRISPRdb (default: '/usr/local/share/CRISPRCasFinder/Repeat_List.csv')
      -DIRrepeat or -drpt [XXX]     Option allowing to use a file file containing repeat IDs and orientation according to CRISPRDirection (default: '/usr/local/share/CRISPRCasFinder/repeatDirection.tsv')
      -flank or -fl [XXX]   Option allowing to set size of flanking regions in base pairs (bp) for each analyzed CRISPR array (default: 100)
      -levelMin or -lMin [XXX]      Option allowing to choose the minimum evidence-level corresponding to CRISPR arrays we want to display (default:1)

      [Detection of Cas clusters]
      -cas or -cs   Search corresponding Cas genes using Prokka (default kingdom: "Bacteria") and MacSyFinder (default: 0)
      -ccvRep or -ccvr      Option used to write the CRISPR-Cas vicinity report (CRISPRs and Cas) if option -cas is set (default: 0)
      -vicinity or -vi [XXX]        Option used to define number of nucleotides separating a CRISPR array from its neighboring Cas system (default: 600)
      -CASFinder or -cf [XXX]       Option allowing to use a custom CasFinder instead of using the CasFinder provided by Institut Pasteur  (default: '/usr/local/share/macsyfinder/CasFinder-2.0')
      -cpuMacSyFinder or -cpuM [XXX]        Option allowing to set number of CPUs to use for MacSyFinder (default: 1)
      -rcfowce      Option allowing to run Casfinder only when any CRISPR exists (default: 0) (set if -cas is set)
      -definition or -def [XXX]     Option allowing to specify CasFinder definition (if option -cas is set) to be more or less stringent (allowed values: 'General', 'Typing' or 'SubTyping'; default: 'General')
      -gffAnnot or -gff [XXX]       Option allowing user to provide an annotation GFF file (if options -cas and -faa are set) (default: '')
      -proteome or -faa [XXX]       Option allowing user to provide a proteome file '.faa' (if options -cas and -gff are set) (default: '')
      -cluster or -ccc [XXX]        Option allowing to constitute clusters or groups of CRISPR or Cas systemes given a determined threshold e.g. 20kb (default: 0) (set if -cas is set)
      -getSummaryCasfinder or -gscf Option allowing to get summary file of Cas-finder (MacSyFinder) and copy it to TSV repository (default: 0)
      -geneticCode or -gcode [XXX]  Option allowing to modify the genetic code (translation table) for CDS annotation (default: 11)

      [Use Prokka instead of Prodigal (default option)]
      -useProkka or -prokka Option allowing to use Prokka instead of Prodigal (default: 0)
      -cpuProkka or -cpuP [XXX]     Option allowing to set number of CPUs to use for Prokka (default: 1)
      -metagenome or -meta  Option allowing to better analyze metagenome with Prokka (default: )
      -ArchaCas or -ac      same option as -cas using "Archaea" as default kingdom instead of "Bacteria" (default: 0). Option to be used when -prokka is used.

    Options waiting for a given parameter (filename, text, or number) are followed by symbols "[XXX]". Other options could be considered as booleans (yes or no, 1 or 0).

    #####################################################

    The input file should meet these constraints:

    - the file name must not contain multiple dots (an acceptable file name is e.g. "multifasta.fna")
    - the sequence must be identified/named (the ID follows character "&gt;", and a description could be added after a space character),
    - the ID should not contain special characters such as "|$%" or multiple dots,
    - the file must contain nucleotides (not amino acids),
    - the file could contain several sequences in FASTA format,
    - each ID must be unique,
    - the ID and the file name must not be too long,
    - the ID will be used for output.

    Examples:
    (1): ./CRISPRCasFinder.simg test.fasta
    In this example, your result folder will be in the directory named: "Result_test"

    (2): ./CRISPRCasFinder -in test.fasta -md 20 -t 33.3 -mr 23 -xr 55 -ms 25 -xs 60 -pm 0.6 -px 2.5 -s 60

    (3): ./CRISPRCasFinder -in genomes100.fna -drpt my_repeatDirection.tsv -rpts my_Repeat_List.csv -cs -fr -dbc my_CRISPR_crisprdb.csv -html

    (4): ./CRISPRCasFinder -in metagenome.fna -rcfowce -prokka -log -out Results_metagenome -cpuProkka 8 -cpuMacSyFinder 8 -meta

    (5): ./CRISPRCasFinder -in sequence.fasta -cas -log -out RES_Sequence -def G -force

%environment
    LC_ALL='C'
    export LC_ALL

%setup
    mkdir -p ${SINGULARITY_ROOTFS}/usr/local/src/CRISPRCasFinder

%post
	export DEBIAN_FRONTEND=noninteractive
	apt-get update
	apt-get install -y apt-utils zlib1g-dev make gcc
	# dash is too restricted
	ln -nsf /bin/bash /bin/sh
    # to be runnable on tars @ Institut Pasteur
    mkdir /pasteur

    apt-get update -y
    apt-get install -y curl default-jre python perl parallel cpanminus patch wget unzip

    ###################
    # Bioinfo package #
    ###################
    apt-get install -y \
    hmmer \
    emboss emboss-lib \
    ncbi-blast+ \
    bioperl \
    bioperl-run \
    libdatetime-perl \
    libxml-simple-perl \
    libdigest-md5-perl \
    clustalw \
    muscle \
    prodigal \
    aragorn \
    infernal \

    cd /usr/bin
    ln -s clustalw2 clustalw2
    cd /

    cpanm Try::Tiny
    cpanm Test::Most
    cpanm JSON::Parse
    cpanm Date::Calc
    cpanm Class::Struct
    cpanm Bio::DB::Fasta
    cpanm File::Copy
    cpanm Bio::Seq Bio::SeqIO
    cpanm --force Bio::Tools::Run::Alignment::Clustalw
    cpanm --force Bio::Tools::Run::Alignment::Muscle

    prefix="/usr/local"

    ##########
    # vmatch #
    ##########
    PN="vmatch"
    PV="2.3.0"
    P="${PN}-${PV}"
    P_SRC=${prefix}/src/${PN}

    mkdir -p ${prefix}/src/vmatch
    cd ${prefix}/src/vmatch
    distribution='Linux_x86_64'
    vmatch="${PN}-${PV}-${distribution}-64bit"
    vmatch_url="http://vmatch.de/distributions/${vmatch}.tar.gz"
    curl -L -O --silent "${vmatch_url}"
    tar -zxf ${vmatch}.tar.gz
    cd ${vmatch}
    gcc -Wall -Werror -fPIC -O3 -shared SELECT/sel392.c -m64 -o sel392v2.so
    # copy the shared library in LD_LIBRARY_PATH
    install -m 0775 sel392v2.so /.singularity.d/libs/sel392v2.so
    cd /.singularity.d/libs/
    ln -s sel392v2.so sel392.so
    cd ${prefix}/src/${PN}/${vmatch}
    install -m 0775 vmatch ${prefix}/bin/vmatch2
    install -m 0775 vsubseqselect ${prefix}/bin/vsubseqselect2
    install -m 0775 mkvtree ${prefix}/bin/mkvtree2
    cd /

    ###############
    # macsyfinder #
    ###############
    PN="macsyfinder"
    PV="1.0.5"
    P="${PN}-${PV}"
    P_SRC=${prefix}/src/${PN}

    mkdir -p ${prefix}/src/${PN}
    cd ${prefix}/src/${PN}
    macsyfinder_url="https://dl.bintray.com/gem-pasteur/MacSyFinder/${P}.tar.gz"
    curl -L -O --silent "${macsyfinder_url}"
    tar -xzf ${P}.tar.gz
    cd ${P}
    python setup.py build
    python setup.py install
    cd /

    #######################
    # prokka dependencies #
    #######################

    ###########
    # signalp #
    ###########

    # Cannot be installed due to Licensing problem.

    ###########
    # tbl2asn #
    ###########
    # trusty package ncbi-tools-bin provide a too old tbl2asn
    PN="tbl2asn"
    PV="1.12"
    P="${PN}-${PV}"
    P_SRC=${prefix}/src/${PN}

    mkdir -p ${P_SRC}
    cd ${prefix}/src/tbl2asn
    tbl2asn_url="ftp://ftp.ncbi.nih.gov/toolbox/ncbi_tools/converters/by_program/${PN}/linux64.${PN}.gz"
    wget "${tbl2asn_url}"
    gunzip linux64.tbl2asn.gz
    install -m 0755 linux64.tbl2asn ${prefix}/bin/${PN}

    ##########
    # prokka #
    ##########
    PN="prokka"
    PV="1.12"
    P="${PN}-${PV}"
    P_SRC=${prefix}/src/${PN}

    mkdir -p ${P_SRC}
    cd ${P_SRC}

    prokka_url="http://www.vicbioinformatics.com/${P}.tar.gz"
    curl -L -O --silent "${prokka_url}"
    tar -xzf ${P}.tar.gz
    cd ${P}

    prokka_data=${prefix}/share/${PN}
    prokka_db=${prokka_data}/db
    test -d ${prokka_db} || mkdir -p ${prokka_db}
    # copy database
    cp -pr db/* ${prokka_db}

    # tell prokka where to find its tools and db once installed
    sed -i -e "s|my \$BINDIR.*|my \$BINDIR=\"${prefix}/libexec/prokka\";|" \
           -e "s|my \$DBDIR.*|my \$DBDIR=\"${prokka_db}\";|" \
           bin/prokka

    for bin in bin/*;
    do
        install -m 0755 ${bin} ${prefix}/bin/
    done

    # install prokka binaries
    test -d ${prefix}/libexec/${PN} || mkdir -p ${prefix}/libexec/${PN}

    for p in binaries/linux/*;
    do
        install -m 0755 ${p} ${prefix}/libexec/${PN}
    done
    # parallel is installed via packet manager
    install -m 0755 binaries/common/minced ${prefix}/libexec/${PN}/
    install -m 0644 binaries/common/minced.jar ${prefix}/libexec/${PN}/

    # setup prokka db
    prokka_cmd="${prefix}/bin/${PN}"

    ${prokka_cmd} --setupdb
    cd /

    ###################
    # CRISPRCasFinder #
    ###################
    PN="CRISPRCasFinder"

    test -d "${prefix}/src/${PN}" || mkdir -p "${prefix}/src/${PN}"
    cd "${prefix}/src/${PN}"

    cripsr_cas_url="https://crisprcas.i2bc.paris-saclay.fr/Home/DownloadFile?filename=CRISPRCasFinder.zip"
    curl -L -o "${PN}.zip" --silent "${cripsr_cas_url}"

    unzip "${PN}.zip"

    cd "${PN}"
    crispr_data="${prefix}/share/${PN}"
    test -d "${crispr_data}" || mkdir "${crispr_data}"

    curl -sLO "https://raw.githubusercontent.com/bneron/CRISPRCasFinder/f10ae5d5306adf7633b1a82e018bbeb5f6119cfb/CRISPRCasFinder.patch"
    patch CRISPRCasFinder.pl CRISPRCasFinder.patch
    curl -sLO "https://raw.githubusercontent.com/bneron/CRISPRCasFinder/e2b9882e5715b1a702fcdc8940a625cbabf49447/singularity/CRISPRCasFinder.singularity.patch"
    patch CRISPRCasFinder.pl CRISPRCasFinder.singularity.patch

    install -m 0755 CRISPRCasFinder.pl ${prefix}/bin/CRISPRCasFinder
    install -m 0644 supplementary_files/crispr.css ${crispr_data}
    install -m 0644 supplementary_files/Repeat_List.csv ${crispr_data}
    install -m 0644 supplementary_files/CRISPR_crisprdb.csv ${crispr_data}
    install -m 0644 supplementary_files/repeatDirection.tsv ${crispr_data}

    #############
    # CasFinder #
    #############
    # use the CasFinder distributed with CRISPRCasFinder
    cas_data="${prefix}/share/macsyfinder/"
    # remove profiles and definitions packaged with macsyfinder
    rm -Rf "${cas_data}DEF"
    rm -Rf "${cas_data}profiles"
    # install cas profiles and definition packaged with CRISPRCasFinder
    cp -r CasFinder-2.0.2 ${cas_data}
    cd /


%test
    stamp=$(date '+%Y-%m-%d-%H:%M:%S')
    result_dir="/tmp/test_CRISPRCasFinder_${stamp}"
    prefix=/usr/local
    crispr_cas_src="${prefix}/src/CRISPRCasFinder/CRISPRCasFinder/"

    CRISPRCasFinder -def General -cas -i "${crispr_cas_src}/install_test/sequence.fasta" -out "${result_dir}" -keep

    returncode=$?
    if [ ${returncode} -ne 0 ];
    then
        echo "Test failed see ${result_dir} for details."
        exit ${returncode}
    fi

    for f in "Cas_REPORT.tsv" "Crisprs_REPORT.tsv";
    do
        diff "${crispr_cas_src}/install_test/${f}" "${result_dir}/TSV/${f}"
        returncode=$?
        if [ ${returncode} -ne 0 ];
        then
            echo "Test failed see ${result_dir} for details."
            exit ${returncode}
        fi
    done

    rm -Rf "${result_dir}"
    exit 0

%runscript
    exec /usr/local/bin/CRISPRCasFinder "$@"
</code></pre>

<h2 id="collection">Collection</h2>

<ul>
  <li>Name: <a href="https://github.com/SciNetHPC/singularity-CRISPRCasFinder">SciNetHPC/singularity-CRISPRCasFinder</a></li>
  <li>License: None</li>
</ul>




<br>
<a class="btn btn-primary" href="https://datasets.datalad.org?dir=/shub/SciNetHPC/singularity-CRISPRCasFinder/latest/2021-01-07-e848fe84-5faa7d00/" target="_blank" type="button">View on Datalad</a>

<h2>Metrics</h2>

<table>
<tr>
  <th>key</th>
  <th>value</th>
</tr>
<tbody>
  <tr>
    <td>id</td>
    <td>/containers/SciNetHPC-singularity-CRISPRCasFinder-latest</td>
  </tr>
  <tr>
    <td>collection name</td>
    <td>SciNetHPC/singularity-CRISPRCasFinder</td>
  </tr>
  <tr>
    <td>branch</td>
    <td>master</td>
  </tr>
  <tr>
    <td>tag</td>
    <td>latest</td>
  </tr>
  <tr>
    <td>commit</td>
    <td>e848fe848233ae1cbdb6a641a8b362a51a3dbc36</td>
  </tr>
  <tr>
    <td>version (container hash)</td>
    <td>5faa7d00f4af74b25dc46341032d51fa</td>
  </tr>
  <tr>
    <td>build date</td>
    <td>2021-01-07T17:33:39.985Z</td>
  </tr>
  <tr>
    <td>size (MB)</td>
    <td>2884</td>
  </tr>
  <tr>
    <td>size (bytes)</td>
    <td>845565983</td>
  </tr>
  <tr>
    <td>SIF</td>
    <td><a href="https://datasets.datalad.org/shub/SciNetHPC/singularity-CRISPRCasFinder/latest/2021-01-07-e848fe84-5faa7d00/5faa7d00f4af74b25dc46341032d51fa.simg">Download URL (please use pull with shub://)</td>
  </tr>
  <tr>
    <td>Datalad URL</td>
    <td><a href="https://datasets.datalad.org?dir=/shub/SciNetHPC/singularity-CRISPRCasFinder/latest/2021-01-07-e848fe84-5faa7d00/">View on Datalad</td>
  </tr>
  <tr>
    <td>Singularity Recipe</td>
    <td><a href="https://datasets.datalad.org/shub/SciNetHPC/singularity-CRISPRCasFinder/latest/2021-01-07-e848fe84-5faa7d00/Singularity">Singularity Recipe on Datalad</td>
  </tr>
</tbody>
</table>
  
<small>We cannot guarantee that all containers will still exist on GitHub.</small>

<script
  src="/singularityhub-archive/assets/js/jquery-3.3.1/jquery-3.3.1.min.js"
  integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8="
  crossorigin="anonymous"></script>

<script>
$(document).ready(function() {

    var toc = $('#TOC');

    // Select each header
    sections = $('.td-content h1');
        $.each(sections, function(idx, v) {
            section = $(v);
            var div_id = $(section).attr('id');
            var div_text = section.text().split('¶')[0];
            var parent = $("#" + div_id)
            var content = '<li id="link_' + div_id + '" class="md-nav__item"><a class="md-nav__link" href="#' + div_id + '" title="' + div_text +'">' + div_text +'</a></li>';
            $(toc).append(content);

            // Add section code to subnavigation
            var children = $('<nav class="md-nav"><ul class="md-nav__list"></nav></ul>')
            var contenders = $("#" + div_id).nextUntil("h1");
            $.each(contenders, function(idx, contender){
               if($(contender).is('h2') || $(contender).is('h3')) {
                   var contender_id = $(contender).attr('id');
                   var contender_text = $(contender).text().split('¶')[0];
                   var content = '<li class="md-nav__item"><a class="md-nav__link" href="#' + contender_id + '" title="' + contender_text +'">' + contender_text +'</a></li>';
                   children.append(content);
                }
             })
             $("#link_" + div_id).append(children);
        });
    });
</script>

<script>
var headers = ["h1", "h2", "h3", "h4"]
var colors = ["red", "orange", "green", "blue"]

$.each(headers, function(i, header){
    var color = colors[i];
    $(header).each(function () {
        var href=$(this).attr("id");
        $(this).append('<a class="headerlink" style="color:' + color + '" href="#' + href + '" title="Permanent link">¶</a>')
    });
})
</script>

<script>
$('h1').first().append('<div><span style="font-size:12px" class="badge badge-danger"><a style="cursor:pointer; color:white" href="https://singularityhub.github.io/singularityhub-archive/tags#latest ">latest</a></span></div>')</script>

	
              
              <style>
  .feedback--answer {
    display: inline-block;
  }
  .feedback--answer-no {
    margin-left: 1em;
  }
  .feedback--response {
    display: none;
    margin-top: 1em;
  }
  .feedback--response__visible {
    display: block;
  }
</style>
<h5 class="feedback--title">Feedback</h5>
<p class="feedback--question">Was this page helpful?</p>
<button class="feedback--answer feedback--answer-yes">Yes</button>
<button class="feedback--answer feedback--answer-no">No</button>
<p class="feedback--response feedback--response-yes">
  Glad to hear it! Please <a href="https://github.com/singularityhub/singularityhub-archive/issues/new">tell us how we can improve</a>.
</p>
<p class="feedback--response feedback--response-no">
  Sorry to hear that. Please <a href="https://github.com/singularityhub/singularityhub-archive/issues/new">tell us how we can improve</a>.
</p>
<script>
  const yesButton = document.querySelector('.feedback--answer-yes');
  const noButton = document.querySelector('.feedback--answer-no');
  const yesResponse = document.querySelector('.feedback--response-yes');
  const noResponse = document.querySelector('.feedback--response-no');
  const disableButtons = () => {
    yesButton.disabled = true;
    noButton.disabled = true;
  };
  const sendFeedback = (value) => {
    if (typeof ga !== 'function') return;
    const args = {
      command: 'send',
      hitType: 'event',
      category: 'Helpful',
      action: 'click',
      label: window.location.pathname,
      value: value
    };
    ga(args.command, args.hitType, args.category, args.action, args.label, args.value);
  };
  yesButton.addEventListener('click', () => {
    yesResponse.classList.add('feedback--response__visible');
    disableButtons();
    sendFeedback(1);
  });
  noButton.addEventListener('click', () => {
    noResponse.classList.add('feedback--response__visible');
    disableButtons();
    sendFeedback(0);
  });
</script><br/>


           </div>
          </main>
        </div>
      </div>
      <footer class="bg-dark py-5 row d-print-none">
  <div class="container-fluid mx-sm-5">
    <div class="row">
      <div class="col-6 col-sm-4 text-xs-center order-sm-2">
        <ul class="list-inline mb-0">  
          
            <li class="list-inline-item mx-2 h3" data-toggle="tooltip" data-placement="top" title="" aria-label="Twitter" data-original-title="Twitter">
              <a class="text-white" target="_blank" href="https://twitter.com/vsoch">
                <i class="fab fa-twitter"></i>
              </a>
            </li>
          
          
            <li class="list-inline-item mx-2 h3" data-toggle="tooltip" data-placement="top" title="" aria-label="LinkedIn" data-original-title="LinkedIn">
              <a class="text-white" target="_blank" href="https://linkedin.com/in/vsochat">
                <i class="fab fa-linkedin"></i>
              </a>
            </li>
          
        </ul>
      </div>
      <div class="col-6 col-sm-4 text-right text-xs-center order-sm-3">
        <ul class="list-inline mb-0">  
          <li class="list-inline-item mx-2 h3" data-toggle="tooltip" data-placement="top" title="" aria-label="GitHub" data-original-title="GitHub">
            <a class="text-white" target="_blank" href="https://github.com/singularityhub/singularityhub-archive">
              <i class="fab fa-github"></i>
            </a>
          </li>
        </ul>
      </div>
      <div class="col-12 col-sm-4 text-center py-2 order-sm-2">
        <small class="text-white">© 2021 Dinosaur Avocado All Rights Reserved</small>
        
        <p class="mt-2"><a href="/about/">About the Archive</a></p>	
      </div>
    </div>
  </div>
</footer>

    </div>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js" integrity="sha384-ZMP7rVo3mIykV+2+9J3UJ46jBk0WLaUAdn689aCwoqbBJiSnjAK/l8WvCWPIPm49" crossorigin="anonymous"></script>
<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js" integrity="sha384-ChfqqxuZUCnJSK3+MXmPNIyE6ZbWh2IMqE241rYiqJxyMiZ6OW/JmZQ5stwEULTy" crossorigin="anonymous"></script>
<script src="/singularityhub-archive/assets/js/main.js"></script>

  </body>
</html>
